demos/imagenet_classification_demo/infer_request_callback.hpp

```diff
@@ -0,0 +1,103 @@
+#include <cstdio>
+#include <string>
+#include <functional>
+#include <atomic>
+
+#include <inference_engine.hpp>
+
+#include <ie_iextension.h>
+
+void resizeImage(cv::Mat& image, int modelInputResolution) {
+    double scale = static_cast<double>(modelInputResolution) / std::min(image.cols, image.rows);
+    cv::resize(image, image, cv::Size(), scale, scale);
+
+    cv::Rect imgROI;
+    if (image.cols >= image.rows) {
+        int fromWidth = image.cols/2 - modelInputResolution/2;
+        imgROI = cv::Rect(fromWidth,
+                          0,
+                          std::min(modelInputResolution, image.cols - fromWidth),
+                          modelInputResolution);
+    }
+    else {
+        int fromHeight = image.rows/2 - modelInputResolution/2;
+        imgROI = cv::Rect(0,
+                          fromHeight,
+                          modelInputResolution,
+                          std::min(modelInputResolution, image.rows - fromHeight));
+    }
+    image(imgROI).copyTo(image);
```

Comment:
This is questionable: the first loop over a data set will involve resizes and crops, but the following will not. The performance will be different.

Reply:
Done

Reply:
I think the issue is still here. The solution can be calling `cv::resize` with new cv::Mat as a second argument within `resizeImage()` and returning that new `cv::Mat` from `resizeImage()`.

Reply:
@Wovchena 
Even though I pass image into the resizeImage() by reference, I use a copy of one of the dataset images every time I call resizeImage(). So I just modify the copy.

cv::Mat nextImage = inputImages[nextImageIndex]; <--- making a copy
resizeImage(nextImage, modelInputResolution);
... and then I use only nextImage

That's why I think that this problem is solved.

Reply:
You can just `return resizedImage(imgROI);` without copy.